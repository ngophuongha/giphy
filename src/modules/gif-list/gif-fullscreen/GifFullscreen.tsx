import { Modal } from "react-bootstrap";
import { GifFullscreenProps } from "./GifFullscreen.type";

export const GifFullscreen = ({
  open,
  setShow,
  imageUrl,
  imageAlt = "gif-alt",
}: GifFullscreenProps): JSX.Element => {
  return (
    <Modal
      dialogClassName="modal-45w"
      aria-labelledby="contained-modal-title-vcenter"
      data-testid="gif-fullscreen"
      centered
      show={open}
      onHide={setShow}
    >
      <img alt={imageAlt} src={imageUrl} />
    </Modal>
  );
};
