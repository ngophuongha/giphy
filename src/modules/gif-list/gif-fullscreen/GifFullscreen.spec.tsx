import { render } from "@testing-library/react";
import { GifFullscreen } from "./GifFullscreen";

describe("Gif Fullscreen", () => {
  it("should render without crashing", () => {
    const { container } = render(
      <GifFullscreen
        open={true}
        setShow={jest.fn}
        imageUrl="http://localhost:8000"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
