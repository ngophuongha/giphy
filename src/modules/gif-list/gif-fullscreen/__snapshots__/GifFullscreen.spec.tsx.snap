// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Fullscreen should render without crashing 1`] = `null`;
